compiler/rustc_infer/src/infer/opaque_types.rs

```diff
@@ -473,17 +473,6 @@ where
                 }
             }
 
-            ty::Alias(ty::Projection, proj) if self.tcx.is_impl_trait_in_trait(proj.def_id) => {
```

Comment:
The change on this line is otherwise identical to 9547302677f1dbd3694b4a477eeee7318d242f4c, except gated behind `lower_impl_trait_in_trait_to_assoc_ty` so it can land before the migration PR.